src/TestableIO.System.IO.Abstractions.TestingHelpers/MockFileSystem.cs

```diff
@@ -322,6 +334,16 @@ public void AddDirectory(string path)
                 var s = StringOperations.EndsWith(fixedPath, separator) ? fixedPath : fixedPath + separator;
                 SetEntry(s, new MockDirectoryData());
             }
+
+            lock (drives)
```

Comment:
I think you should move this logic to the `SetEntry` method. Currently you would automatically create a drive only when you add a new Directory, but not when you add a new File.

Reply:
Done.

I also moved the `lock (files)` into `SetEntry`; however, I'm not sure if this is valid. What do you think?

Reply:
LGTM